ts/components/services/NewServiceListItem.tsx

```diff
@@ -51,99 +35,27 @@ const styles = StyleSheet.create({
-        switchValue
-      });
-    }
-  }
-
-  private getServiceKey = (potService: pot.Pot<ServicePublic, Error>): string =>
-    pot.getOrElse(
-      pot.map(
-        potService,
-        service => `${service.service_id}-${service.version || 0}`
-      ),
-      `service-switch`
-    );
-
-  private isInboxChannelEnabled() {
-    const potService = this.props.item;
-    const uiEnabledChannels = pot.map(potService, service =>
-      getEnabledChannelsForService(this.props.profile, service.service_id)
-    );
-    return pot.isSome(uiEnabledChannels) && uiEnabledChannels.value.inbox;
-  }
-
-  private onItemSwitchValueChanged(value: boolean) {
-    const potService = this.props.item;
-    if (this.props.onItemSwitchValueChanged !== undefined) {
-      const onItemSwitchValueChanged = this.props.onItemSwitchValueChanged;
-      pot.map(potService, service => {
-        // if the service is not updating and the new value is
-        // different from the old one, then update it!
-        if (
-          !pot.isUpdating(this.props.profile) &&
-          this.state.switchValue !== value
-        ) {
-          this.setState({
-            switchValue: value
-          });
-          onItemSwitchValueChanged([service], value);
-        }
-      });
-    }
-  }
-
-  public render() {
-    const { switchValue } = this.state;
-    const potService = this.props.item;
-
-    const onPress = !this.props.isLongPressEnabled
-      ? pot.toUndefined(
-          pot.map(potService, service => () => this.props.onSelect(service))
-        )
-      : undefined;
-
-    const serviceName = pot.isLoading(potService)
-      ? I18n.t("global.remoteStates.loading")
-      : pot.isError(potService) || pot.isNone(potService)
-      ? I18n.t("global.remoteStates.notAvailable")
-      : potService.value.service_name;
-
-    return (
-      <ListItemComponent
-        title={serviceName}
-        hasBadge={false} // disabled for these reasons https://www.pivotaltracker.com/story/show/176919053
-        onPress={onPress}
-        onLongPress={this.props.onLongPress}
-        hideSeparator={this.props.hideSeparator}
-        style={styles.listItem}
-        isItemDisabled={servicesRedesignEnabled ? false : !switchValue}
-        onSwitchValueChanged={this.onItemSwitchValueChanged}
-        switchValue={switchValue}
-        switchDisabled={pot.isUpdating(this.props.profile)}
-        keySwitch={this.getServiceKey(potService)}
-        isLongPressEnabled={this.props.isLongPressEnabled}
-      />
-    );
-  }
-}
+const NewServiceListItem = (props: Props): React.ReactElement => {
+  const potService = props.item;
+  const onPress = pot.toUndefined(
+    pot.map(potService, service => () => props.onSelect(service))
+  );
+
+  const serviceName = pot.isLoading(potService)
+    ? I18n.t("global.remoteStates.loading")
+    : pot.isError(potService) || pot.isNone(potService)
+    ? I18n.t("global.remoteStates.notAvailable")
+    : potService.value.service_name;
```

Comment:
with 4 branches combine in ternaries, I would have preferred a `fold` operation tbh 🤔

Reply:
Good point. Me too https://github.com/pagopa/io-app/pull/3220/commits/5cc0a639c2cbcfd0b1e2f2c530c80ec2aee0e36d

Reply:
more readable fro sure, even if I don't really appreciate the `pot.fold` syntax 😆